caluma/extensions/events/form.py

```diff
@@ -0,0 +1,122 @@
+    """
+    question = instance.answer.question
+    main_document = instance.answer.document
+
+    summary_question = question.meta.get("summary-question")
+    summary_mode = question.meta.get("summary-mode")
+
+    if not summary_question and not summary_mode:
+        # no summary requested
+        return
+
+    msg = f"Updating table summary: tq={instance.answer.question_id}"
+    logger.debug(msg)
+
+    if not summary_question or not summary_mode:
+        msg = f"Updating table summary: missing info in TQ meta: sq={summary_question},sm={summary_mode}"
+        logger.warning(msg)
+        return
+
+    msg = f"Updating table summary: sq={summary_question} sm={summary_mode}"
+    logger.debug(msg)
+
+    summary_answer, _ = caluma_form_models.Answer.objects.get_or_create(
+        document=main_document, question_id=summary_question
+    )
+
+    summary_modes = {"csv": _make_csv_summary}
+
+    summary_func = summary_modes.get(summary_mode)
+    if not summary_func:
+        msg = f'Updating table summary: summary mode "{summary_mode}" does not exist. Must be one of {settings.TABLE_SUMMARY_MODES}'
+        logger.warning(msg)
+        return
+
+    summary_answer.value = summary_func(instance.answer)
+    summary_answer.save()
+
+
+@on(post_save, sender=caluma_form_models.Answer, raise_exception=True)
+@transaction.atomic
+def update_table_summary_from_row(instance, *args, **kwargs):
+    ad = caluma_form_models.AnswerDocument.objects.filter(
+        document=instance.document
+    ).first()
+    if not ad:
+        return
+
+    # AnswerDocument available, we're in a table
+    # Make sure the summaries (if any) are updated
+    update_table_summary(instance=ad)
+
+
+def _make_csv_summary(table_answer):
+    def get_lines(answer_docs, row_question_slugs):
+        for ad in answer_docs:
+            result = {}
+            answer_data = {
+                a.question.slug: a.value
+                for a in ad.document.answers.filter(question_id__in=row_question_slugs)
+            }
+            for question_slug in row_question_slugs:
+                result[question_slug] = answer_data.get(question_slug, "")
+            yield result
+
+    msg = f"Making CSV summary for {table_answer.question}"
+    logger.debug(msg)
+    answer_docs = caluma_form_models.AnswerDocument.objects.filter(
+        answer=table_answer
+    ).order_by("-sort")
+    row_question_slugs = _sorted_form_question_slugs(table_answer.question.row_form)
+
+    with io.StringIO() as csvfile:
+        writer = csv.DictWriter(
+            csvfile,
+            fieldnames=row_question_slugs,
+            delimiter=";",
+            quoting=csv.QUOTE_MINIMAL,
+        )
+        writer.writeheader()
+        for line in get_lines(answer_docs, row_question_slugs):
+            writer.writerow(line)
+
+        result = csvfile.getvalue()
+    msg = f"Making CSV summary for {table_answer.question}: result={result}"
+    logger.debug(msg)
```

Comment:
```suggestion
    logger.debug(f"Making CSV summary for {table_answer.question}: result={result}")
```

... might also consider dropping that log entirely